osu.Game/Rulesets/UI/FrameStabilityContainer.cs

```diff
@@ -85,35 +85,52 @@ protected override void LoadComplete()
 
         public override bool UpdateSubTree()
         {
-            state = frameStableClock.IsPaused.Value ? PlaybackState.NotValid : PlaybackState.Valid;
+            if (parentGameplayClock == null)
+                setClock(); // LoadComplete may not be run yet, but we still want the clock.
+
+            double aimTime = parentGameplayClock.CurrentTime;
+
+            if (frameStableClock.WaitingOnFrames.Value)
```

Comment:
Am I the only one that can't follow this logic? It's split between 3 methods - updateClock(), updateReplay() and now UpdateSubTree(), all of which directly contribute to the state.

I really don't like this. Why can't this be inside updateClock() at the very least, with a further improvement being to separate it out into its own method that isolates its purpose?

Reply:
can you explain which logic you mean? the fact that this variable is set inside `updateClock`?

Reply:
All of the new logic added in UpdateSubTree is very hard to follow. Including the parent gameplay clock now being set here instead of updateClock()

Reply:
I can attempt to move it back. I initially moved it over to attempt to fix the reported issue, but it turns out that was not required.